### PR TITLE
Create packages using fpm instead of debuild

### PR DESCRIPTION
**Pull request by JonasGroeger**
This PR introduces a shell script called `build-deb` which can be used to create debian packages of this project.

After sucessfully building the project using `grunt build`, you can then go ahead and run one of

```
VERSION=0.6.4 DIST=linux64 ./fpm/build-deb
VERSION=0.6.4 DIST=linux32 ./fpm/build-deb
```

`DIST=linux64` is the default.

Using [fpm](https://github.com/jordansissel/fpm) makes it also easier to build `rpm`s or other packages.

Since fpm does not yet support signing deb packages (see jordansissel/fpm#170, jordansissel/fpm#809, jordansissel/fpm#467), this isn't done as of yet.

### CODE REVIEW

**Comment by jakejarrett** [comment]
This looks awesome, I'll try it out tonight. :smile_cat:

**Comment by JonasGroeger** [comment]
@jakejarrett I have asked [packagecloud.io](https://packagecloud.io/) for an Open-Source repository. Lets see what they answer. This would streamline getting the package to users.

File: fpm/build-deb [hunk 1]
```diff
@@ -56,4 +54,7 @@ fpm -t deb \
     --description "Soundnode App is an Open-Source project to support Soundcloud for desktop Mac, Windows, and Linux." \
     --after-install "fpm/create-symlink" \
     --after-remove "fpm/remove-symlink" \
-    --deb-no-default-config-files
+    --deb-no-default-config-files \
+    "$DIST_DIR/.=/usr/share/soundnode/" \
```
Comment (JonasGroeger) [inline fpm/build-deb]: The app

File: fpm/build-deb [hunk 2]
```diff
@@ -56,4 +54,7 @@ fpm -t deb \
...
+    "app/soundnode.png=/usr/share/pixmaps/" \
```
Comment (JonasGroeger) [inline fpm/build-deb]: The icon

File: fpm/build-deb [hunk 3]
```diff
@@ -56,4 +54,7 @@ fpm -t deb \
...
+    "fpm/soundnode.desktop=/usr/share/applications/"
```
Comment (JonasGroeger) [inline fpm/build-deb]: The .desktop file

File: fpm/build-deb (hunk 2 above)
Comment (jakejarrett) [inline fpm/build-deb]: @JonasGroeger should this be /usr/share/icons, or is this a more distro neutral approach?

Comment (JonasGroeger) [inline fpm/build-deb]: Typically the "icons" folder contains icons related to the Gnome (and KDE I assume, but I have no knowledge of this) themes. The "pixmaps" folder contains icons related to specific applications.

**Comment by JonasGroeger** [comment]
@jakejarrett Ideally, I would have an array of icons (or a svg to convert it myself) in the following sizes:

In `PNG` format:
* 22x22
* 16x16
* 32x32
* 24x24
* 48x48
* 128x128
* 256x256

And additionally a `SVG` file for
* scalable

**Comment by JonasGroeger** [comment]
In the latest commit I had to resize the icon from apps/soundnode.png in 114x114 to fpm/soundnode-48x48.png in 48x48. This means the icon is now essentially duplicated. Maybe we should convert it automatically using
 
    convert -resize 48 app/soundnode.png fpm/soundnode-48x48.png

This would however require imagemagick to be installed for building debian packages. If thats OK @jakejarrett , I can add this step to the `build-deb` script.

File: fpm/build-deb (hunk 2 above)
Comment (JonasGroeger) [inline fpm/build-deb]: @jakejarrett True. I changed that in https://github.com/Soundnode/soundnode-app/pull/777/commits/56e013fd242fc4be0dc504a385e9614f078f7fa1

**Comment by JonasGroeger** [comment]
@jakejarrett They answered telling me that its no problem having an account for that. I created it and will be maintaining a debian repository containing a package of this project. Aside from the small issue with the `*.svg` icon (for which I have also mailed @pmarquees), this is ready to merge imho.

**Comment by jakejarrett** [comment]
@JonasGroeger Looks good, I'd suggest squashing the commits down to a single commit if possible, Besides that, it should be good to merge.

**Comment by JonasGroeger** [comment]
@jakejarrett With delight :)

**Comment by JonasGroeger** [comment]
@jakejarrett ping :)

**Comment by jakejarrett** [comment]
@JonasGroeger sorry been a bit preoccupied with some stuff lately. This is good to merge.

**Pull request merged**